src/System.Memory/src/System/Buffers/Text/Utf8Formatter/FormattingHelpers.cs

```diff
@@ -0,0 +1,142 @@
+    {
+        // For the purpose of formatting time, the format specifier contains room for
+        // exactly 7 digits in the fraction portion. See "Round-trip format specifier"
+        // at the following URL for more information.
+        // https://msdn.microsoft.com/en-us/library/az4se3k1(v=vs.110).aspx#Roundtrip
+        private const int FractionDigits = 7;
+
+        // A simple lookup table for converting numbers to hex.
+        private const string HexTable = "0123456789abcdef";
+
+        #region UTF-8 Helper methods
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void WriteHexByte(byte value, ref byte buffer, int index)
+        {
+            Unsafe.Add(ref buffer, index) = (byte)HexTable[value >> 4];
+            Unsafe.Add(ref buffer, index + 1) = (byte)HexTable[value & 0xF];
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int WriteFractionDigits(long value, int digitCount, ref byte buffer, int index)
+        {
+            for (int i = FractionDigits; i > digitCount; i--)
+                value /= 10;
+
+            return WriteDigits(value, digitCount, ref buffer, index);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int WriteDigits(long value, int digitCount, ref byte buffer, int index)
+        {
+            long left = value;
+
+            for (int i = digitCount - 1; i >= 0; i--)
+            {
+                left = DivMod(left, 10, out long num);
+                Unsafe.Add(ref buffer, index + i) = (byte)('0' + num);
+            }
+
+            return digitCount;
+        }
+
+        /// <summary>
+        /// The unsigned long implementation of this method is much slower than the signed version above
+        /// due to optimization tricks that happen at the IL to ASM stage. Use the signed version unless
+        /// you definitely need to deal with numbers larger than long.MaxValue.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int WriteDigits(ulong value, int digitCount, ref byte buffer, int index)
+        {
+            ulong left = value;
+
+            for (int i = digitCount - 1; i >= 0; i--)
+            {
+                left = DivMod(left, 10, out ulong num);
+                Unsafe.Add(ref buffer, index + i) = (byte)('0' + num);
+            }
+
+            return digitCount;
+        }
+
+        #endregion UTF-8 Helper methods
+
+        #region Math Helper methods
+
+        /// <summary>
+        /// We don't have access to Math.DivRem, so this is a copy of the implementation.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static long DivMod(long numerator, long denominator, out long modulo)
+        {
+            long div = numerator / denominator;
+            modulo = numerator - (div * denominator);
+            return div;
```

Comment:
This is already the implementation of Math.DivRem in .NET Core:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/shared/System/Math.cs#L82-L90
```
            long div = a / b;
            result = a - (div * b);
            return div;
```